src/avatar_action.cpp

```diff
@@ -334,7 +334,7 @@ bool avatar_action::move( avatar &you, map &m, int dx, int dy, int dz )
         if( you.has_effect( effect_riding ) && you.mounted_creature != nullptr ) {
             auto mon = you.mounted_creature.get();
             if( !mon->has_flag( MF_SWIMS ) || mon->get_size() < you.get_size() + 2 ) {
-                add_msg( m_warning, _( "Your mount shies away from the water!" ) );
+                add_msg( m_warning, _( "The creature you are riding cannot swim while it is carrying you!" ) );
```

Comment:
Let's give it a name:
```suggestion
                add_msg( m_warning, _( "The %s cannot swim while it is carrying you!" ), mon->get_name() );
```